STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/_az_nx_blob_client.c

```diff
@@ -0,0 +1,148 @@
+#include "_az_nx_blob_client.h"
+
+#define USER_AGENT_NAME "User-Agent: "
+#define USER_AGENT_VALUE "Azure RTOS Device (STM32)"
+
+static az_result result_from_nx_status(UINT nx_status)
+{
+  switch (nx_status)
+  {
+    case NX_SUCCESS:
+      return AZ_OK;
+    case NX_WEB_HTTP_POOL_ERROR:
+      return AZ_ERROR_NOT_ENOUGH_SPACE;
+    case NX_NO_PACKET:
+      return AZ_ERROR_NOT_ENOUGH_SPACE;
+    case NX_INVALID_PARAMETERS:
+      return AZ_ERROR_NOT_SUPPORTED;
+    case NX_OPTION_ERROR:
+      return AZ_ERROR_ARG;
+    case NX_PTR_ERROR:
+      return AZ_ERROR_ARG;
+    case NX_CALLER_ERROR:
+      return AZ_ERROR_ARG;
+    case NX_WEB_HTTP_NOT_READY:
+      return AZ_ERROR_ULIB_BUSY;
+    case NX_WAIT_ABORTED:
+      return AZ_ERROR_ULIB_SYSTEM; // should be AZ_ERROR_ULIB_TIME_OUT
+                                   // or AZ_ERROR_ULIB_ABORTED
+    default:
+      return AZ_ERROR_ULIB_SYSTEM;
+  }
+}
+
+az_result az_nx_blob_client_init(
+    NXD_ADDRESS* ip,
+    NX_WEB_HTTP_CLIENT* http_client,
+    NX_PACKET* packet_ptr,
+    CHAR* resource,
+    CHAR* host,
+    ULONG wait_option)
+{
+  UINT nx_status;
+  az_result result = AZ_OK;
+
+  // create http blob client
+  if ((nx_status = nx_web_http_client_create(http_client, "HTTP Client", &nx_ip, &nx_pool, 1536))
+      != NX_SUCCESS)
+  {
+    result = result_from_nx_status(nx_status);
+  }
+
+  // connect to server
+  else if (
+      (nx_status
+       = nx_web_http_client_connect(http_client, ip, NX_WEB_HTTP_SERVER_PORT, wait_option))
+      != NX_SUCCESS)
+  {
+    result = result_from_nx_status(nx_status);
+  }
+
+  // initialize get request
+  else if (
+      (nx_status = nx_web_http_client_request_initialize(
+           http_client,
+           NX_WEB_HTTP_METHOD_GET, /* GET, PUT, DELETE, POST, HEAD */
+           resource, /* "resource" (usually includes auth headers)*/
+           host, /* "host" */
+           0, /* Used by PUT and POST */
+           NX_FALSE, /* If true, input_size is ignored. */
+           NX_NULL, /* "name" */
+           NX_NULL, /* "password" */
+           wait_option))
+      != NX_SUCCESS)
+  {
+    result = result_from_nx_status(nx_status);
+  }
+
+  // add user agent
+  else if (
+      (nx_status = nx_web_http_client_request_header_add(
+           http_client,
+           USER_AGENT_NAME,
+           sizeof(USER_AGENT_NAME) - 1,
+           USER_AGENT_VALUE,
+           sizeof(USER_AGENT_VALUE) - 1,
+           wait_option))
+      != NX_SUCCESS)
+  {
+    result = result_from_nx_status(nx_status);
+  }
+
+  return result;
```

Comment:
> result; [](http://example.com/codeflow?start=8&length=8)

Instead of having one conversion for each call, consider a single one in the return.
```
return result_from_nx_status(nx_status);
```